cpp/pipelines/remove_dangling_references.cc

```diff
@@ -52,7 +63,7 @@ void EliminateDanglingCrossReferences(MARC::Reader * const reader, MARC::Writer
                 for (const auto &subfield : field->getSubfields()) {
                     if (subfield.code_ == 'w' and StringUtil::StartsWith(subfield.value_, "(DE-627)")) {
                         const auto bsz_ppn(subfield.value_.substr(__builtin_strlen("(DE-627)")));
-                        if (unlikely(all_ppns.find(bsz_ppn) == all_ppns.cend())) {
+                        if (unlikely(all_ppns.find(bsz_ppn) == all_ppns.cend()) or all_ppns.find(bsz_ppn)->second == true) {
```

Comment:
If you would like to use the result of the find() operation more than once, please use a variable to avoid duplicate search operations.

Reply:
Also: Didn't we want to have a different kind of logging if the record is skipped on purpose?